src/domain/bitcoin/onchain/index.ts

```diff
@@ -42,28 +58,14 @@ export const IncomingOnChainTransaction = ({
   rawTx: OnChainTransaction
   fee: Satoshis
   createdAt: Date
-}): IncomingOnChainTransaction =>
-  BaseOnChainTransaction({
-    confirmations,
-    rawTx,
-    fee,
-    createdAt,
-  })
-
-export const OutgoingOnChainTransaction = ({
+}): OutgoingOnChainTransaction => ({
   confirmations,
   rawTx,
   fee,
   createdAt,
-}: {
-  confirmations: number
-  rawTx: OnChainTransaction
-  fee: Satoshis
-  createdAt: Date
-}): OutgoingOnChainTransaction =>
-  BaseOnChainTransaction({
-    confirmations,
-    rawTx,
-    fee,
-    createdAt,
-  })
+  uniqueAddresses: () =>
```

Comment:
a comment to explain what we're doing could be useful. I guess we're trying to get a list of onchain addresses but why do we need a TxOut is is not obvious

Reply:
this is part of a previous implementation. I guess was done to fix a bug with a deposit but really not sure. @bodymindarts do you remember?

Reply:
A transaction has multiple UTXOs - this function returns a list of unique addresses from all the UTXOs in the transaction. Note not every UTXO must send to an address (due to `OP_RETURN`) and some UTXOs could send to the same address (though it would be bad).
Its used to get all wallets that need updating when a tx comes in [here](https://github.com/GaloyMoney/galoy/blob/main/src/app/wallets/update-on-chain-receipt.ts#L40-L41)